src/main/java/io/confluent/idesidecar/restapi/application/SnappyNative.java

```diff
@@ -0,0 +1,48 @@
+package io.confluent.idesidecar.restapi.application;
+
+import io.quarkus.logging.Log;
+import io.quarkus.runtime.StartupEvent;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
```

Comment:
Would +1 a big docstring here explaining the background of why we're having to do this. A new reader will be mighty confused about the overall end-end process as to why this is necessary and what happens if we don't bend so far over backwards.